osbs/api.py

```diff
@@ -696,71 +697,34 @@ def _do_create_prod_build(self, git_uri, git_ref,
                 # fix for module requires will have to be determined from experience.
                 raise ValueError("Flatpak build cannot be isolated")
 
+        if not git_branch:
+            raise OsbsValidationException("required argument 'git_branch' can't be None")
+
         repo_info = utils.get_repo_info(git_uri, git_ref, git_branch=git_branch,
                                         depth=git_commit_depth)
-        build_request = self.get_build_request(inner_template=inner_template,
-                                               outer_template=outer_template,
-                                               customize_conf=customize_conf,
-                                               arrangement_version=arrangement_version)
-
         if flatpak:
             req_labels, base_image = self._get_flatpak_labels(repo_info)
         else:
             req_labels, base_image = self._check_labels(repo_info)
+        kwargs.pop('name_label', None)
+        kwargs.pop('component', None)
+        kwargs.pop('base_image', None)
```

Comment:
Is this necessary? When a python function receives the same kwarg twice, you get a TypeError, e.g.:
```python
>>> def f(a=1, **kwargs):
...     pass

>>> f(a=2, **{'a': 3})
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: f() got multiple values for keyword argument 'a'
```
So there should be no danger of having the intended values overwritten by the content of `kwargs`, and if these arguments are passed to `_do_create_prod_build()`, it  might be better to let it fail rather than silently ignoring them.

Reply:
Just me being lazy rather than fixing the tests properly.